Allow greatest-based tests to use GREATEST_MAIN_BEGIN() without using
GREATEST_MAIN_DEFS() in that same translation unit.

In other words, allow greatest-based tests to assume they will be linked
against other code that does GREATEST_MAIN_DEFS() on their behalf.

This makes it possible to integrate greatest.h with CMake's
create_test_sourcelist() feature, which consolidates multiple tests into
a single executable and thereby reduces the total amount of linking work
incurred by a from-scratch build.

https://cmake.org/cmake/help/latest/command/create_test_sourcelist.html

--- greatest-src/greatest.h.orig	2024-09-24 17:20:51.916897368 -0700
+++ greatest-src/greatest.h	2024-09-24 17:23:07.099252901 -0700
@@ -895,8 +895,8 @@
     report_suite();                                                     \
 }                                                                       \
                                                                         \
-static void greatest_run_suite(greatest_suite_cb *suite_cb,             \
-                               const char *suite_name) {                \
+void greatest_run_suite(greatest_suite_cb *suite_cb,                    \
+                        const char *suite_name) {                       \
     if (greatest_suite_pre(suite_name)) {                               \
         suite_cb();                                                     \
         greatest_suite_post();                                          \
@@ -935,7 +935,7 @@
         name);                                                          \
 }                                                                       \
                                                                         \
-static void greatest_parse_options(int argc, char **argv) {             \
+void greatest_parse_options(int argc, char **argv) {                    \
     int i = 0;                                                          \
     for (i = 1; i < argc; i++) {                                        \
         if (argv[i][0] == '-') {                                        \
@@ -1259,6 +1259,12 @@
 
 #endif /* USE_ABBREVS */
 
+/*
+ * Forward declarations for key subset of GREATEST_MAIN_DEFS()
+ */
+void greatest_parse_options(int, char **);
+void greatest_run_suite(greatest_suite_cb *, const char *);
+
 #if defined(__cplusplus) && !defined(GREATEST_NO_EXTERN_CPLUSPLUS)
 }
 #endif
